### PR TITLE
ARM: 'azure network' commands migrated to Autorest SDK

### PR DESCRIPTION
**Pull request by romangromov**
This PR contains major changes related to most of  `azure network` commands in `ARM`:

* `azure-arm-network` package updated to latest `0.11.0`. Now it's based on Autorest instead of Hydra;
* `--resource-group` parameter now optional, related issue https://github.com/Azure/azure-xplat-cli/issues/2342 fixed;
* code base re-worked to follow single convention;
* test cases updated;
* nock recordings updated;
* unused nocks removed;
* commands manually tested by QA: @sreekanthbs

For visibility: @DeepakRajendranMsft @FrankSiegemund @amarzavery @yugangw-msft

### CODE REVIEW

File: lib/commands/arm/armsdk/network.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//
```
Comment (amarzavery) [inline lib/commands/arm/armsdk/network.js]: Why is this needed if the autorest based network package (0.11.0) has been published?

File: lib/util/utils.js [hunk 2]
```diff
@@ -229,11 +229,20 @@ exports.createComputeResourceProviderClient = function (subscription) {
   return _createArmClient(factoryMethod, subscription);
 };
 
+/**
+ * Deprecated, use createNetworkManagementClient instead.
+ */
 exports.createNetworkResourceProviderClient = function (subscription) {
```
Comment (amarzavery) [inline lib/util/utils.js]: This should go away

File: lib/commands/arm/armsdk/network.js (hunk 1 above)
Comment (romangromov) [inline lib/commands/arm/armsdk/network.js]: @amarzavery it's temporary workaround for `vm compute` commands, because it's not yet migrated. We have a related -email thread about this and I've received confirmation from Pengfei Huang Huang.Pengfei@microsoft.com

File: lib/util/utils.js (hunk 2 above)
Comment (romangromov) [inline lib/util/utils.js]: @amarzavery unfortunately we still need this, because `azure vm` commands not migrated to `azure-arm-network` from autorest

**Comment by romangromov** [comment]
@amarzavery please look over this again, we should be good to merge, thx!

File: lib/util/utils.js [hunk 3]
```diff
@@ -73,17 +73,26 @@ exports.createClient = function (factoryMethod, credentials, endpoint, options)
...
     polishErrorCausedByArmProviderNotRegistered()
-    ];
-  
-  var client = new FactoryMethod(credentials, subscriptionId, exports.stringTrimEnd(endpoint, '/'), options);
+  ];
+
+  var client = new FactoryMethod(credentials, subscriptionId, baseUri, options);
+
+  // To reduce playback time in test cases
+  var isRecording = process.env.AZURE_NOCK_RECORD || false;
```
Comment (yugangw-msft) [inline lib/util/utils.js]: @romangromov @amarzavery , let us do not put the testing workaround in the production code. The right place should be [here](https://github.com/Azure/azure-xplat-cli/blob/dev/test/framework/cli-test.js#L591).

File: lib/commands/arm/armsdk/network.js (hunk 1 above)
Comment (amarzavery) [inline lib/commands/arm/armsdk/network.js]: agreed.

File: lib/util/utils.js (hunk 3 above)
Comment (amarzavery) [inline lib/util/utils.js]: That is correct. Yugang has a right point.

File: lib/util/utils.js (hunk 2 above)
Comment (amarzavery) [inline lib/util/utils.js]: agreed

File: lib/util/utils.js (hunk 3 above)
Comment (romangromov) [inline lib/util/utils.js]: @yugangw-msft thank you, I've add `wrap` for `createArmClient`, please take a look

**Comment by yugangw-msft** [comment]
LGTM overall. @amarzavery , could you spend a little more time in the detail to ensure the signature change caused by autorest is expected. Once good, just merge

**Comment by romangromov** [comment]
@amarzavery

**Comment by romangromov** [comment]
@amarzavery
